cardstack/src/components/TransactionConfirmationSheet/displays/components/sections/MerchantSectionCard.tsx

```diff
@@ -34,18 +40,20 @@ const MerchantSectionCard = ({
             textColor={merchantInfoDID?.textColor}
           />
         ) : (
-          <Icon name="user" size={80} />
+          customIcon
+        )}
+        {!isPaymentReceived && (
+          <Container paddingTop={3} marginBottom={4}>
+            <Text
+              weight="extraBold"
+              size="medium"
+              ellipsizeMode="tail"
+              numberOfLines={1}
+            >
+              {merchantInfoDID?.name || ''}
+            </Text>
+          </Container>
```

Comment:
We can check just for merchantInfoDID, if `merchantInfoDID` exists we render the name, otherwise we don't, so no need for `isPaymentReceived` flag.

```suggestion
        {!!merchantInfoDID && (
          <Container paddingTop={3} marginBottom={4}>
            <Text
              weight="extraBold"
              size="medium"
              ellipsizeMode="tail"
              numberOfLines={1}
            >
              {merchantInfoDID?.name || ''}
            </Text>
          </Container>
```

Reply:
nice catch! just missed it